src/include/sof/audio/volume.h

```diff
@@ -94,18 +114,24 @@ typedef uint32_t (*vol_zc_func)(const struct audio_stream *source,
 
 typedef int32_t (*vol_ramp_func)(struct comp_dev *dev, int32_t ramp_time, int channel);
 
-/**
- * \brief Volume component private data.
- *
- * Gain amplitude value is between 0 (mute) ... 2^16 (0dB) ... 2^24 (~+48dB).
- */
 struct vol_data {
+#if CONFIG_IPC_MAJOR_4
+	struct ipc4_base_module_cfg base;	/**< module config */
+	uint32_t mailbox_offset;		/**< store peak volume in mailbox */
+
+	/**< these values will be stored to mailbox for host (IPC4). In case
+	  *  of IPC3 it is only to store volume, target volume, peak values
+	  */
+	struct ipc4_peak_volume_regs peak_regs;
```

Comment:
@abonislawski for my education, is it correct that in IPC4 these values will be written to the mailbox after every copy()? What is the intention for this? Is there any expectation from the host driver to do something with these?

Reply:
Not sure really, it was recently added also to copier, maybe for debug or driver will print this in specific circumstances